tools/data/textrecog/mtwi_converter.py

```diff
@@ -0,0 +1,256 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+import json
+import math
+import os
+import os.path as osp
+
+import mmcv
+from PIL import Image
+
+from mmocr.datasets.pipelines.crop import crop_img
+from mmocr.utils.fileio import list_to_file
+
+
+def collect_files(img_dir, gt_dir, ratio):
+    """Collect all images and their corresponding groundtruth files.
+    Args:
+        img_dir (str): The image directory
+        gt_dir (str): The groundtruth directory
+        ratio (float): Split ratio for val set
+
+    Returns:
+        files (list): The list of tuples (img_file, groundtruth_file)
+    """
+    assert isinstance(img_dir, str)
+    assert img_dir
+    assert isinstance(gt_dir, str)
+    assert gt_dir
+    assert isinstance(ratio, float)
+    assert ratio < 1.0, 'val_ratio should be a float between 0.0 to 1.0'
+
+    ann_list, imgs_list = [], []
+    for ann_file in os.listdir(gt_dir):
+        img_file = osp.join(img_dir, ann_file.replace('txt', 'jpg'))
+        # This dataset contains some images obtained from .gif,
+        # which cannot be loaded by mmcv.imread(), convert them
+        # to RGB mode.
+        if mmcv.imread(img_file) is None:
```

Comment:
- In my case, an image named `TB1PTJiLXXXXXaeaFXXunYpLFXX.jpg` will always fail to be decompressed when conducting `unzip -q ...`. The error information says that ` invalid compressed data to inflate image train/TB1PTJiLXXXXXaeaFXXunYpLFXX.jpg`
- There should be some content in this image according to its corresponding annotation. However when I check this image. there is noting in it and it is a blank file.
- And an error will raise at line 38 when conducting imread here: `cv2.error: OpenCV(4.5.4) /tmp/pip-req-build-jpmv6t9_/opencv/modules/imgcodecs/src/loadsave.cpp:814: error: (-215:Assertion failed) !buf.empty() in function 'imdecode_'`
- Currently I just remove this image and its annotation, then everything will be fine. 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Reply:
1. I didn't encounter such an issue on my side. But I added a cv2.error exception, please check if it works.
2. Can you also try to run the code on the dataset zip I shared on the server to see if the error still exists? So, we can find out if it is a dataset zip issue or code issue.

Reply:
ok